src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/MSBuildProcessManager.cs

```diff
@@ -142,6 +127,87 @@ void Process_OutputDataReceived(object sender, DataReceivedEventArgs e)
                     output.AppendLine(e.Data);
                 }
             }
+
+            async Task<ProcessResult> GetTimeoutForProcess(Process process, TimeSpan? timeout)
+            {
+                await Task.Delay(timeout.Value);
+
+                // Don't timeout during debug sessions
+                while (Debugger.IsAttached)
+                {
+                    Thread.Sleep(TimeSpan.FromSeconds(1));
+                }
+                if (!process.HasExited)
+                {
+                    var procDumpProcess = await CaptureDump(process);
+                    if (procDumpProcess != null && procDumpProcess.HasExited)
+                    {
+                        Console.WriteLine("ProcDump failed to run.");
+                        procDumpProcess.Kill();
+                    }
+
+                    // This is a timeout.
+                    process.Kill();
+                }
+
+                throw new TimeoutException($"command '${process.StartInfo.FileName} {process.StartInfo.Arguments}' timed out after {timeout}. Output: {output.ToString()}");
+            }
+
+            async Task<Process> CaptureDump(Process process)
+            {
+                var metadataAttributes = Assembly.GetExecutingAssembly()
+                    .GetCustomAttributes<AssemblyMetadataAttribute>();
+
+                var procDumpPath = metadataAttributes
+                    .SingleOrDefault(ama => ama.Key == "ProcDumpPath")?.Value;
+
+                if (string.IsNullOrEmpty(procDumpPath))
+                {
+                    Console.WriteLine("ProcDumpPath not defined.");
+                    return null;
+                }
+                var procDumpExePath = Path.Combine(procDumpPath, "procdump.exe");
+                if (!File.Exists(procDumpExePath))
+                {
+                    Console.WriteLine($"Can't find procdump.exe in '{procDumpPath}'.");
+                    return null;
+                }
+
+                var dumpDirectory = metadataAttributes
+                    .SingleOrDefault(ama => ama.Key == "ArtifactsLogDir")?.Value;
+
+                if (string.IsNullOrEmpty(dumpDirectory))
+                {
+                    Console.WriteLine("ArtifactsLogDir not defined.");
+                    return null;
+                }
+
+                if (!Directory.Exists(dumpDirectory))
+                {
+                    Console.WriteLine($"'{dumpDirectory}' does not exist.");
```

Comment:
Why not create the directory?

Reply:
I want to avoid this causing errors at runtime that mask the actual failure of the test. The folder is created at build time, if not, something is wrong..

Reply:
Make this silent. Should do nothing unless the file exists.

Reply:
Same comment as above regarding the logging.